optuna/structs.py

```diff
@@ -85,18 +83,67 @@ class FrozenTrial(
         user_attrs:
             Dictionary that contains the attributes of the :class:`~optuna.trial.Trial` set with
             :func:`optuna.trial.Trial.set_user_attr`.
-        system_attrs:
-            Dictionary that contains the attributes of the :class:`~optuna.trial.Trial` internally
-            set by Optuna.
         intermediate_values:
             Intermediate objective values set with :func:`optuna.trial.Trial.report`.
-        trial_id:
-            Optuna's internal identifier of the :class:`~optuna.trial.Trial`. Note that this field
-            is not supposed to be used by library users. Instead, please use :attr:`number` and
-            :class:`~optuna.study.Study.study_id` to identify a :class:`~optuna.trial.Trial`.
     """
 
-    internal_fields = ['distributions', 'trial_id']
+    def __init__(
+        self,
+        number,  # type: int
+        state,  # type: TrialState
+        value,  # type: Optional[float]
+        datetime_start,  # type: Optional[datetime]
+        datetime_complete,  # type: Optional[datetime]
+        params,  # type: Dict[str, Any]
+        distributions,  # type: Dict[str, BaseDistribution]
+        user_attrs,  # type: Dict[str, Any]
+        system_attrs,  # type: Dict[str, Any]
+        intermediate_values,  # type: Dict[int, float]
+        trial_id,  # type: int
+    ):
```

Comment:
Please add a type hint for `__init__`'s return value like `# type: (...) -> None`. I think it cannot be omitted as mentioned in [PEP484](https://www.python.org/dev/peps/pep-0484/#the-meaning-of-annotations).

Reply:
Added as suggested. Wonder why it passes `mypy` though.